ddtrace/_trace/utils_botocore/aws_payload_tagging.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Optional
+import json
+import copy
+
+from decimal import Decimal
+from ddtrace import Span
+from ddtrace import config
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.constants import SPAN_KIND
+from ddtrace.constants import SPAN_MEASURED_KEY
+from ddtrace.ext import SpanKind
+from ddtrace.ext import aws
+from ddtrace.ext import http
+from ddtrace.internal.constants import COMPONENT
+from ddtrace.internal.utils.formats import deep_getattr
+from ddtrace.propagation.http import HTTPPropagator
+from ddtrace.vendor.jsonpath_ng import parse
+
+MAX_TAGS = 758 # RFC-defined maximum number of allowed tags
+INCOMPLETE_TAG = "_dd.payload_tags_incomplete" # Set to True if MAX_TAGS is reached
+
+class AWSPayloadTagging:
+    def __init__(self):
+        # TODO is this _total_ tags for the span or is this _total_ expanded tags that we added?
+        # TODO is this a limit on the backend or something?
+        # currently treating this as the maximum number of tags we will add from the JSON
+        self.current_tag_count = 0
+
+    def expand_payload_as_tags(self, span: Span, result: Dict[str, Any], key: str) -> None:
+        """
+        Expands the JSON payload from various AWS services into tags and sets them on the Span.
+        """
+        if not result:
+            return
+
+        redacted_dict = copy.deepcopy(result)
+        self._redact_json(redacted_dict, span)
+
+        for key2, value in redacted_dict.items():
+            self._tag_object(span, f"{key}.{key2}", value)
+            if self.current_tag_count >= MAX_TAGS:
+                return
+
+    def _should_try_string(self, obj: Any) -> bool:
+        try:
+            if isinstance(obj, str) or isinstance(obj, unicode):
+                return True
+        except NameError:
+            if isinstance(obj, bytes):
+                return True
+
+        return False
+
+    def _redact_json(self, data: Dict[str, Any], span: Span) -> None:
+        """
+        Redact sensitive data in the JSON payload based on default and user-provided JSONPath expressions
+        """
+        # TODO should we cache these paths and generated lists?
```

Comment:
This is me not knowing python very well, the `Node.js` implementation uses a special JSON file to store these and read from these, Java doesn't.

Reply:
Nope, after they are configured they are set, I attempted to cache them.

I don't think I can use `@cached` here as the `data` will always be different?